src/main/scala/vectorpipe/ProcessOSM.scala

```diff
@@ -211,13 +228,34 @@ object ProcessOSM {
     * @param elements DataFrame containing node, way, and relation elements
     * @return DataFrame containing geometries.
     */
-  def constructGeometries(elements: DataFrame): DataFrame = {
-    import elements.sparkSession.implicits._
+  def constructGeometries(input: DataFrame): DataFrame = {
+    import input.sparkSession.implicits._
+    val st_pointToGeom = org.apache.spark.sql.functions.udf { pt: jts.Point => pt.asInstanceOf[jts.Geometry] }
+
+    // The following type conversion needed in case input comes from Change source
+    val elements =
+      Try(input.schema
+               .find(_.name == "members")
+               .get
+               .dataType
+               .asInstanceOf[ArrayType]
+               .elementType
+               .asInstanceOf[StructType]
+               .find(_.name == "type")
+               .get) match {
+        case Failure(_) => throw new IllegalArgumentException(s"Could not get type of 'members' in schema ${input.schema}")
+        case Success(field) =>
+          if (field.dataType == ByteType)
+            input.withColumn("members", convertMembers('members))
+          else
+            input
+      }
```

Comment:
I would do this conversion the other way to use `Byte`s internally rather than strings. Other internal bits should already use bytes.

Reply:
Huh.  So, why was I seeing errors when passing in dataframes that used Bytes in the `members` field?

Reply:
Ah, because `preprocessRelations` *also*  compresses member types (converts them to `Byte`s): https://github.com/geotrellis/vectorpipe/pull/60/files#diff-edada4a44d234744c962d1de3348a36eL197

Reply:
Egad.  I should then pull out this conversion for broader use?  This sort of hinges on knowing what the top-level functions are, I guess.  To know which functions should call the conversion?

Reply:
In retrospect, I should have used `Byte` types in `osm2orc`, but I still like the idea of it being a 1:1 match to the PBF version. However, it's pretty wasteful storing all of those duplicate strings in a DataFrame (I think; it's not for an ORC file due to the dictionary encoding), especially when the shuffle hits 1TB at points, so my preference would be to try to use `Byte` types internally (and generate them, e.g. via the streaming sources) while also transparently converting them from the ORC versions.

What do you think?

Reply:
> I should then pull out this conversion for broader use?

Yeah, probably. Blerg.